src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -245,22 +249,26 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                 BuildFallbackBlock(builder);
             }
 
-            // We've taken over tag rendering, so prevent rendering the outer tag
-            output.TagName = null;
-            output.Content.SetContent(builder);
+            output.PostElement.SetContent(builder);
         }
 
         private void BuildGlobbedLinkTags(TagHelperAttributeList attributes, TagHelperContent builder)
         {
             EnsureGlobbingUrlBuilder();
 
-            // Build a <link /> tag for each matched href as well as the original one in the source file
-            var urls = GlobbingUrlBuilder.BuildUrlList(Href, HrefInclude, HrefExclude);
+            // Build a <link /> tag for each matched href.
+            var urls = GlobbingUrlBuilder.BuildUrlList(null, HrefInclude, HrefExclude);
             foreach (var url in urls)
             {
                 // "url" values come from bound attributes and globbing. Must always be non-null.
                 Debug.Assert(url != null);
 
+                if (string.Equals(Href, url, StringComparison.OrdinalIgnoreCase))
+                {
+                    // Don't build script tag for the original source url.
```

Comment:
Since the original `href` attribute is no longer passed into `BuildUrlList()`, this comment seems incorrect. Is this actually about protecting against duplicates of the `href` value in the globbed results? Suspect this block isn't needed.